### PR TITLE
Add TOTP MFA management custom flow

### PR DESCRIPTION
**Pull request by royanger**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1194/custom-flows/manage-app-based-mfa

### Explanation: 
Custom flow code for managing application based multi-factor authentication

### This PR:
Documentation was missing

### CODE REVIEW

File: docs/custom-flows/manage-app-based-mfa.mdx [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  const generateBackupCodes = () => {
+    setLoading(true);
+    void user
+      ?.createBackupCode()
+      .then((backupCodes: BackupCodeResource) => {
+        setBackupCodes(backupCodes);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.log('Error:', error);
```
Comment (alexisintech) [inline docs/custom-flows/manage-app-based-mfa.mdx:69]: Do we want to add this for error handling?

```suggestion
        // See https://clerk.com/docs/custom-flows/error-handling
        // for more info on error handling
        console.error(JSON.stringify(err,null,2));
```

File: docs/manifest.json [hunk 2]
```diff
@@ -750,6 +750,10 @@
                         {
                           "title": "User impersonation",
                           "href": "/docs/custom-flows/user-impersonation"
+                        },
+                        {
+                          "title": "Manage MFA",
```
Comment (royanger) [inline docs/manifest.json:755]:
```suggestion
                          "title": "Manage TOTP-based MFA",
```

File: docs/custom-flows/manage-totp-mfa.mdx [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+<Tabs items={["Manage MFA page", "Add TOTP page"]}>
+  <Tab>
+    ```jsx {{ filename: '/app/account/manage-mfa/page.tsx' }}
+    'use client';
+
+    import * as React from 'react';
+    import { useUser } from '@clerk/nextjs';
+    import Link from 'next/link';
+    import { BackupCodeResource, UserResource } from '@clerk/types';
```
Comment (royanger) [inline docs/custom-flows/manage-totp-mfa.mdx:45]:
```suggestion
    import { BackupCodeResource } from '@clerk/types';
```

File: docs/custom-flows/manage-totp-mfa.mdx [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    }
+    ```
+  </Tab>
+
+  <Tab>
+    ```jsx {{ filename: '/app/account/manage-mfa/add/page.tsx' }}
+    'use client';
+
+    import { useUser } from '@clerk/nextjs';
+    import { BackupCodeResource, TOTPResource, UserResource } from '@clerk/types';
```
Comment (royanger) [inline docs/custom-flows/manage-totp-mfa.mdx:168]:
```suggestion
    import { BackupCodeResource, TOTPResource } from '@clerk/types';
```

File: docs/custom-flows/manage-totp-mfa.mdx [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+      );
+    }
+
+    export default function AddMFaScreen() {
+      const [step, setStep] = React.useState<AddTotpSteps>('add');
+      const { isLoaded, user } = useUser();
+
+      if (!isLoaded) return null;
+
+      if (isLoaded && !user) {
```
Comment (royanger) [inline docs/custom-flows/manage-totp-mfa.mdx:335]:
```suggestion
      if (!user) {
```
This should be redundant with the `!isLoaded` check above.

File: docs/custom-flows/manage-totp-mfa.mdx [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+            <Link href="/account/manage-mfa/add">
+              <button>Add</button>
+            </Link>
+          </p>
+        </div>
+      );
+    };
+
+    // Generate and display backup codes
+    function GenerateBackupCodes() {
```
Comment (royanger) [inline docs/custom-flows/manage-totp-mfa.mdx:80]: In the original code, I had this as some basic logic in the `ManageTOTP` (now `ManageMFA`). Now that this is is own component, its a near duplicate to the `DisplayBackupCodes` component from the `/app/account/manage-mfa/add/page.tsx` section.

Let's either revert this to the more basic version from the original code, or use one component (this one probably) and import it on the other page.

**Comment by royanger** [comment]
Changes look good. This is ready to merge.

**Review by alexisintech — APPROVED** [review]
WOO!!

**Pull request merged**